h2/src/main/org/h2/mode/PgCatalogTable.java

```diff
@@ -81,6 +81,16 @@
 
     private static final int META_TABLE_TYPE_COUNT = PG_USER + 1;
 
+    private static final Object[][] PG_EXTRA_TYPES = {
+            { 18, "char", 1, 0 },
+            { 19, "name", 64, 18 },
+            { 22, "int2vector", -1, 21 },
+            { 24, "regproc", 4, 0 },
+            { PgServer.PG_TYPE_INT2_ARRAY, "_int2", -1, PgServer.PG_TYPE_INT2 },
+            { PgServer.PG_TYPE_INT4_ARRAY, "_int4", -1, PgServer.PG_TYPE_INT4 },
```

Comment:
I think all array types should be grouped together. `VARCHAR ARRAY` can be defined there too and code in lines 556–572 can skip `Value.ARRAY` to avoid duplicate definition.